lib/content-services/src/lib/search/components/search-input/search-input.component.ts

```diff
@@ -0,0 +1,114 @@
+/*!
+ * @license
+ * Copyright © 2005-2023 Hyland Software, Inc. and its affiliates. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { CommonModule } from '@angular/common';
+import { Component, EventEmitter, Input, Output, ViewEncapsulation } from '@angular/core';
+import { MatFormFieldModule } from '@angular/material/form-field';
+import { MatInputModule } from '@angular/material/input';
+import { TranslateModule } from '@ngx-translate/core';
+
+@Component({
+    selector: 'app-search-input',
+    standalone: true,
+    imports: [CommonModule, MatFormFieldModule, MatInputModule, TranslateModule],
+    templateUrl: `./search-input.component.html`,
+    styleUrls: ['./search-input.component.scss'],
+    encapsulation: ViewEncapsulation.None
+})
+export class SearchInputComponent {
+    @Input()
+    value = '';
+
+    @Input()
+    label = 'SEARCH.INPUT.LABEL';
+
+    @Input()
+    placeholder = 'SEARCH.INPUT.PLACEHOLDER';
+
+    @Input()
+    fields: string[] = ['cm:name'];
+
+    @Output()
+    changed = new EventEmitter<string>();
+
+    onSearchInputChanged(event: Event) {
+        const input = event.target as HTMLInputElement;
+        const searchTerm = input.value;
+
+        const query = this.formatSearchQuery(searchTerm, this.fields);
+        if (query) {
+            this.changed.emit(decodeURIComponent(query));
+        }
+    }
+
+    private formatSearchQuery(userInput: string, fields = ['cm:name']) {
+        if (!userInput) {
+            return null;
+        }
+
+        if (/^http[s]?:\/\//.test(userInput)) {
+            return this.formatFields(fields, userInput);
+        }
+
+        userInput = userInput.trim();
+
+        if (userInput.includes(':') || userInput.includes('"')) {
+            return userInput;
+        }
+
+        const words = userInput.split(' ');
+
+        if (words.length > 1) {
+            const separator = words.some(this.isOperator) ? ' ' : ' AND ';
+
+            return words
+                .map((term) => {
+                    if (this.isOperator(term)) {
```

Comment:
I think it can be done in one line like ` return this.isOperator(term) ? term : this.formatFields(fields, term);`